lib/internal/fs/rimraf.js

```diff
@@ -42,6 +42,34 @@ const separator = Buffer.from(sep);
 function rimraf(path, options, callback) {
   let retries = 0;
 
+  if (options.recursiveStrict) {
+    if (stat(path, (err, stats) => {
```

Comment:
I think we need `stat` vs `lstat` because we actually want to follow the symlink but I'm not 100% sure.

Reply:
I would check the behavior of `rmdir` on a symlink, without recursive, and just match it.